crepecake/src/main/java/com/google/cloud/tools/crepecake/image/Image.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.crepecake.image;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Represents an image.
+ *
+ * @param <T> the type of {@link Layer} this image contains
+ */
+public class Image<T extends Layer> {
```

Comment:
Maybe I don't understand what an image is, but can't it be different kinds of layers?

Reply:
Yea, if the image has different kinds of layers, it would just be Image<Layer>. However, for certain images, like the base image or a built-but-not-pushed image, it might only consist of a certain type of layer. Perhaps templating this not the best way to approach it. I'll remove the template on this for now.

Reply:
Well if you're using it to enforce certain paradigms that might be fine.

Reply:
I agree with removing the generics.